zipkin-lens/src/components/DependenciesPage/DependenciesGraph.tsx

```diff
@@ -0,0 +1,312 @@
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+/* eslint-disable no-shadow */
+import React, { CSSProperties } from 'react';
+import ReactSelect, { ValueType, ActionMeta } from 'react-select';
+import { AutoSizer } from 'react-virtualized';
+import { Box, Grid, useTheme } from '@material-ui/core';
+import { createStyles, makeStyles, Theme } from '@material-ui/core/styles';
+import moment from 'moment';
+
+import Dependencies from '../../types/Dependencies';
+import VizceralWrapper from './VizceralWrapper';
+import NodeDetailData from './NodeDetailData';
+import { Edge } from './types';
+
+// These filter functions use any type because they are passed directly to untyped JS code.
+const filterConnections = (object: any, value: any) => {
+  if (!value) {
+    return true;
+  }
+  if (object.name === value) {
+    return true;
+  }
+  return object.source.name === value || object.target.name === value;
+};
+
+const filterNodes = (object: any, value: any) => {
+  if (!value) {
+    return true;
+  }
+  if (object.name === value) {
+    return true;
+  }
+  return (
+    object.incomingConnections.find(
+      (conn: any) => conn.source.name === value,
+    ) ||
+    object.outgoingConnections.find((conn: any) => conn.target.name === value)
+  );
+};
+
+// Export for testing.
+export const getNodesAndEdges = (dependencies: Dependencies) => {
+  const nodes = [] as { name: string }[];
+  const edges = [] as Edge[];
+
+  dependencies.forEach((edge) => {
+    const nodeNames = nodes.map((node) => node.name);
+
+    if (!nodeNames.includes(edge.parent)) {
+      nodes.push({ name: edge.parent });
+    }
+    if (!nodeNames.includes(edge.child)) {
+      nodes.push({ name: edge.child });
+    }
+
+    edges.push({
+      source: edge.parent,
+      target: edge.child,
+      metrics: {
+        normal: edge.callCount || 0,
+        danger: edge.errorCount || 0,
+      },
+    });
+  });
+  return { nodes, edges };
+};
+
+const reactSelectStyles = {
+  control: (base: CSSProperties) => ({
+    ...base,
+    width: '15rem',
+  }),
+  option: (base: CSSProperties) => ({
+    ...base,
+    cursor: 'pointer',
+  }),
+  clearIndicator: (base: CSSProperties) => ({
+    ...base,
+    cursor: 'pointer',
+  }),
+  dropdownIndicator: (base: CSSProperties) => ({
+    ...base,
+    cursor: 'pointer',
+  }),
+};
+
+const useStyles = makeStyles((theme: Theme) =>
+  createStyles({
+    containerGrid: {
+      height: '100%',
+    },
+    graphItemGrid: {
+      height: '100%',
+    },
+    nodeDetailDataItemGrid: {
+      height: '100%',
+      boxShadow: theme.shadows[10],
+      zIndex: 1,
+    },
+    vizceralWrapper: {
+      // In Firefox, 100% height does not work...
+      '& .vizceral': {
+        height: '99%',
+      },
+    },
+  }),
+);
+
+interface Props {
```

Comment:
Let's name `Prop` interfaces e.g. `DependenciesGraphProps`. While in apps, it's not so common to export / import props interfaces, it still happens and it prevents having to then rename it at the importer.